### ISSUE TITLE
Cudnn version of the problem

### ISSUE THREAD

**Issue by zhangshengsheng** [tensorflow/tensorflow#23715]
Loaded runtime CuDNN library: 7.1.2 but source was compiled with: 7.3.1.  CuDNN library major and minor version needs to match or have higher minor version in case of CuDNN 7.0 or later version. If using a binary install, upgrade your CuDNN library.  If building from sources, make sure the library loaded at runtime is compatible with the version specified during compile configuration.

 cat /usr/local/cuda/include/cudnn.h | grep CUDNN_MAJOR -A 2
#define CUDNN_MAJOR 7
#define CUDNN_MINOR 4
#define CUDNN_PATCHLEVEL 1
--
#define CUDNN_VERSION (CUDNN_MAJOR * 1000 + CUDNN_MINOR * 100 + CUDNN_PATCHLEVEL)

#include "driver_types.h"

No 7.1.2 installed.

 tf-nightly-gpu  1.13.0.dev20181110

 tensorflow-gpu  1.12.0rc2

**Comment by PK15946** [comment]
I have exactly the same error, do you have any progress yet? @wt-huang

**Comment by jjyunlp** [comment]
amazing .... check carefully whether there is a 7.1.12 cudnn. Maybe you have only updated the cudnn.h but left the old libcudnn.so?

**Comment by zubairahmed-ai** [comment]
In Windows 10 I am getting the same error though CuDNN has been updated to 7.3.1 for CUDA 9
What could be the issue?

**tensorflow/stream_executor/cuda/cuda_dnn.cc:363] Loaded runtime CuDNN library: 7.0.5 but source was compiled with: 7.2.1.  CuDNN library major and minor version needs to match or have higher minor version in case of CuDNN 7.0 or later version.**

**Comment by zubairahmed-ai** [comment]
@abdel In my case I had two installations of CUDA on different drives and I was copying cuDNN files to the wrong one initially, so it kepts loading old binaries and gave me the same error until I copied to the right one, make sure you do the same

**Comment by abdel** [comment]
@zubairahmed-ai Thanks for the tip. There's definitely only one CUDA in our case on a single drive, and all PATHs seem to indicate it should be using that one, even CUDA/cuDNN test samples confirm that.. yet TensorFlow seems to be getting 7.0.5 from somewhere...?

**Comment by ZacMonroe** [comment]
@abdel I am having the exact same problem.

**Comment by ZacMonroe** [comment]
Installing cuDNN 7.1.4 seemed to fix my problem.

**Comment by acotterson** [comment]
I found a fix on my system! On my Windows computer using Anaconda, I found that using conda for some package (tensorflow? although I'm successfully using a custom tensorflow package) automatically grabbed another version of cudnn as a dependency--the version that was giving me troubles. Updating through conda and pip wouldn't give me a version more recent, so I fixed it through manually updating the cudnn package. I went to my install of Anaconda3 and went here:

C:\Anaconda3\pkgs\cudnn-7.1.4-cuda9.0_0\Library

In here, you find a bin, include, and lib folder. Using the exact same process that was used installing cudnn with cuda, i replaced the cudnn files in these folders with the ones from the version I wanted. 
I also went to the library path of my environment

C:\Anaconda3\envs\ai1\Library\lib\x64

and replaced this cudnn.lib file as well.

Hopefully this makes sense and it helps somebody!

**Comment by jvishnuvardhan** [comment]
I think it was resolved. I am closing the issue. Please open new ticket if you see similar issue again. Thanks!

**Comment by XYudong** [comment]
add `export LD_LIBRARY_PATH=/usr/local/cuda/lib64:${LD_LIBRARY_PATH:+${LD_LIBRARY_PATH}}` to the ~/.bashrc file, and then `source ~/.bashrc`. This solves my problem!

tf:1.12, CUDA 9.0, 384, cuDNN.7.1.4

**Comment by Pidem** [comment]
I had a similar issue working with TF2. Turns out this was due to the fact that I was using an old conda environment that didn't have access to the same cuDNN.

**Comment by wm2012011492** [comment]
I have exactly the same error, do you have any solution now?

**Comment by mpodlasin** [comment]
I also have this error - can we reopen? :(

**Comment by jvishnuvardhan** [comment]
@mpodlasin Please open a new issue with your issue details and error trace. Thanks!

**Comment by Corly** [comment]
I also had this error on Windows. On my case I had installed cupy-coda which also added de cudnn dll and the program always looked there for it.

In order to fix this I just removed the cupy-coda dll from the cupy-cuda install folder (..Python36\Lib\site-packages\cupy\cuda) and then the program loaded the correct cudnn dll from CUDA_PATH.

**Comment by berylsheep-up** [comment]
I solve the same solution by follow method.

I add the follow values into the System environment variable.
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\libnvvp
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\bin
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\lib
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\include

**Comment by ameicler** [comment]
Downgrading to a compatible version (from 1.15 to 1.14) of TF for my CUDA version solved it for me.

Compatbility table : https://www.tensorflow.org/install/source#linux

**Comment by oerton42** [comment]
> I also had this error on Windows. On my case I had installed cupy-coda which also added de cudnn dll and the program always looked there for it.
> 
> In order to fix this I just removed the cupy-coda dll from the cupy-cuda install folder (..Python36\Lib\site-packages\cupy\cuda) and then the program loaded the correct cudnn dll from CUDA_PATH.

You made my day !!!
Tensorflow did not find cudnn after installiong cupy.


1) install tensorflow (using conda install keras-gpu is easy(
2) install cupy-conda101 or another wheel as explains the cupy website
3) uninstall cudnn.dll from your anaconda env
4) make sure you have the correct version of cudnn installed and your windows path correctly entered

**Comment by chandu8542** [comment]
I was having similar issue. I had an old version of cudnn (8.0.4) somewhere in `/usr/lib` and `LD_LIBRARY-PATH` was set in `.bashrc` and somehow still referring the old version of cudnn. If you re getting this error, check if you have an old version of cudnn somewhere.  So what I did was added a `conf` file in `/etc/ld.so.conf.d/ and did `sudo ldconfig` so that tf can load cudnn8.2.x . Surpringly, my code is running super fast.
